### PR TITLE
Add switch expression to UCR

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Part of an effort to switch all the abt reports over to static configurations. This will make the data source configurations _much_ easier to maintain.

@czue cc: @snopoke

### CODE REVIEW

File: corehq/apps/userreports/expressions/factory.py [hunk 1]
```diff
@@ -35,6 +35,16 @@ def _conditional_expression(spec, context):
     return wrapped
 
 
+def _switch_expression(spec, context):
+    wrapped = SwitchExpressionSpec.wrap(spec)
+    wrapped.configure(
+        ExpressionFactory.from_spec(wrapped.switch_on, context),
+        {k : ExpressionFactory.from_spec(v, context) for k, v in wrapped.cases.iteritems()},
```
Comment (dimagimon) [inline corehq/apps/userreports/expressions/factory.py]: E203 whitespace before ':'

File: corehq/apps/userreports/README.md [hunk 2]
```diff
@@ -146,6 +147,41 @@ Note that this expression contains other expressions inside it! This is why expr
...
+            "type": "constant",
+            "constant": 65
+        },
+    },
+    "default": {
+        "type": "constant",
+        "constant": 0
+    }
+}
+```
```
Comment (czue) [inline corehq/apps/userreports/README.md]: nice

**Comment by czue** [comment]
:+1: when tests pass

**Comment by czue** [comment]
this looks great - though in the back of my mind i'm wondering at what point we are just writing a new programming language :smiling_imp:

**Comment by NoahCarnahan** [comment]
Yeah, I've been thinking about that too. I wonder how hard it would be to replace expressions with javascript functions, sort of the way couch uses js for views.

**Comment by NoahCarnahan** [comment]
Maybe something like this: https://github.com/asvd/jailed

**Comment by czue** [comment]
neat - yeah i agree that could be useful.

**Comment by NoahCarnahan** [comment]
tests passed

**Pull request merged**